src/mono/mono/mini/interp/transform.c

```diff
@@ -3655,10 +3661,10 @@ interp_transform_call (TransformData *td, MonoMethod *method, MonoMethod *target
 			}
 #endif
 		}
-		td->last_ins->flags |= INTERP_INST_FLAG_CALL;
 	}
 	td->ip += 5;
-	td->last_ins->info.call_args = call_args;
+	init_last_ins_call (td);
+	td->last_ins->info.call_info->call_args = call_args;
```

Comment:
The `call_args` field was incorrectly set here. We should only init in places where `INTERP_INST_FLAG_CALL` is set, aka above

Reply:
In `interp_transform_call` there is https://github.com/dotnet/runtime/blob/main/src/mono/mono/mini/interp/transform.c#L3556-L3577 that doesn't set `INTERP_INST_FLAG_CALL` which is probably correct. I will add there allocation of `InterpCallInfo` structure.